Add basic control flow

lowjiayee opened the pull request:
Refer to commit message `3e733d1` for details

Please do not follow the structure of `AttendanceCommand`. Should follow the structure from classes under package `seedu.command.event`.

lowjiayee committed:
Add basic io

lowjiayee committed:
Add parseEvent (only for adding event)

lowjiayee committed:
Update code style

lowjiayee committed:
Remove IO to another branch

lowjiayee committed:
Merge branch 'master' into JiaYee-Parser

lowjiayee committed:
Modify interpreter logic (work-in-progress)

lowjiayee committed:
Add basic control flow

Start from...
1.Duke
2.CommandInterpreter
3.EventCommandInterpreter
4.Add
5.EventList

lowjiayee committed:
Improve readability of classes and functions

lowjiayee committed:
Merge branch 'master' into JiaYee-Parser

harithadiv commented on src/main/java/seedu/parser/Parser.java:
```diff
@@ -1,21 +1,109 @@
+                    throw new UnknownFormatFlagsException("unknown flag");
+                }
+                if (mostRecent == null) {
+                    throw new NullPointerException("parameter without flag");
```
is it better to throw all exceptions as DukeExceptions and use try catch blocks for specific exception? 
ie. catch(NullPointerException e){
      throw new DukeException(e.getMessage());
  }

lowjiayee replied:
Ya, but I let all of them throw DukeException instead. I have also refactored the checks into individual functions to improve readability.

lowjiayee committed:
Fix incorrect override of Command.execute()

Take note that execute() for Attendance and Performance related commands are deleted to conform to overriding of abstract execute() method.

lowjiayee committed:
Ensure consistent exception handling

lowjiayee committed:
Edit text-ui-test
